server-connection/src/main/proto/sonarlint.proto

```diff
@@ -45,25 +45,26 @@ message ProjectComponents {
 }
 
 message AnalyzerConfiguration {
-    map<string, string> settings = 1;
-    map<string, RuleSet> ruleSetsByLanguageKey = 2;
+  map<string, string> settings = 1;
+  map<string, RuleSet> rule_sets_by_language_key = 2;
+  uint32 schema_version = 3;
 }
 
 message RuleSet {
-    repeated ActiveRule rules = 1;
-    string lastModified = 3;
-
-    message ActiveRule {
-        string ruleKey = 1;
-        string severity = 2;
-        string templateKey = 3;
-        map<string,string> params = 4;
-    }
+  repeated ActiveRule rule = 1;
```

Comment:
As mentioned in the commit comment, despite Google's code style being to make repeated fields plural, we decided long time ago to have them singular, in order to have better Java method names. This field was the only exception. I don't mind changing to make all repeated fields plural, but this is something a bit more impactful (this can change method names used externally), so I would prefer to align first in the team, and do that not in a patch release.